reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -74,85 +83,116 @@ object StreamSubscriber {
     def onFinalize: F[Unit]
 
     /** producer for downstream */
-    def dequeue1: F[Either[Throwable, Option[A]]]
+    def dequeue1: F[Chunk[Either[Throwable, Option[A]]]]
```

Comment:
We could name this `dequeueChunk` so it's clear by name what it's doing.

Reply:
or `dequeueChunk1`

Reply:
That's true. Will fix.

Reply:
Not essential, but it might be worth seeing if we can reduce the amount of boxing/allocations here especially given we know we'll get exactly one of `onError` or `onComplete`, i.e. I think we can define
```scala
def dequeueChunk: F[Either[Throwable, Option[Chunk[A]]]]
```